lib/pbench/server/api/resources/__init__.py

```diff
@@ -618,6 +641,40 @@ def __init__(
         self.schema = schema
         self.role = role
 
+    def _validate_query_params(self, request: Request, schema: Schema) -> JSONOBJECT:
+        """
+        When an API accepts HTTP query parameters from the URL, these aren't
+        automatically validated by the dispatcher. This method collects query
+        parameters into a JSON object and validates them against a specified
+        schema.
+
+        Args:
+            request:    The HTTP Request object containing query parameters
+            schema:     The Schema definition
+
+        Returns:
+            The resulting JSON object
+        """
+        json = {}
+        badkey = []
+        for key in request.args.keys():
+            if key in schema:
+                values = request.args.getlist(key)
+                if len(values) > 1 and schema[key].type == ParamType.LIST:
+                    json[key] = values
+                else:
+                    json[key] = values[0]
```

Comment:
Why use the first one?  (Why not the last one, or why not reject the key, if it's a list when it's not supposed to be?)

Reply:
The code before used `getlist` for a `LIST`-type parameter and `get` for a non-list parameter. The actual behavior of the Werkzeug `Multidict.get` is to return the first element of the list and I just translated the idiom.

Basically it means that `?name=foo&name=bar` will ignore the extra; and, you're right, maybe we shouldn't do that, but it wasn't a nuance I considered at the time.

Hmm.